include/NovelRT.Interop/Utilities/NrtMisc.h

```diff
@@ -13,8 +13,8 @@ extern "C"
 
     const char* Nrt_getExecutablePath();
     const char* Nrt_getExecutableDirPath();
-    const char* Nrt_appendFilePath(int32_t numberOfArgs, ...);
-    const char* Nrt_appendText(int32_t numberOfArgs, ...);
+    const char* Nrt_appendFilePath(int32_t numberOfArgs, const char* const* args);
```

Comment:
`varargs` are not compatible with .NET and so we need explicit lists instead.

Reply:
I see....is there no way to just have both? Or too much overhead do you thnk?

Reply:
I think its more overhead than its worth. It buys very little over putting the types in a stack allocated buffer, basically just saves a line of code per call.

I think it would be better to define common "overloads" for taking in 2-3 parts. They just need unique names because "C".

Reply:
The underlying impl is variadic though, so ultimately I don't think having multiple "overloads" would really work out too well.

This is probably the best we can do.

Reply:
It works out the same? It's basically a question of forcing the user to do:
```cpp
const char* const pathParts[2] = { part1, part2 };
Nrt_appendFilePath(2, pathParts);
```

or allowing them to do the following, where it does (at the minimum) the above (or possibly something more efficient:
```cpp
Nrt_appendFilePath2(part1, part2);
```

`Nrt_appendText` and `Nrt_appendFilePath` both use `string::append` which is non-variadic: https://www.cplusplus.com/reference/string/string/append/
The same would be true if we used `filesystem::path` for concatenating.

This is similar to how .NET has `string.Format(string format, object arg)`, `string.Format(string format, object arg1, object arg2)`, `string.Format(string format, object arg1, object arg2, object arg3)` and `string.Format(string format, params object[] args)`, because the first three can be more efficient

Reply:
Wait really? Then why are we using variadic for the API surface itself?

Reply:
> Then why are we using variadic for the API surface itself?

Could you clarify?

Reply:
I think Matt was asking why were even using variadic params in the API in the first place. If so, then the answer is - "I didn't know about this at the time 😃"

If this is working as expected, then I'm okay with it. It seems like a simple enough change - varargs can get dicey IIRC anyways.